libpod/sql_state_internal.go

```diff
@@ -15,6 +15,137 @@ import (
 	_ "github.com/mattn/go-sqlite3"
 )
 
+// Checks that the DB configuration matches the runtime's configuration
+func checkDB(db *sql.DB, r *Runtime) (err error) {
+	// Create a table to hold runtime information
+	// TODO: Include UID/GID mappings
+	const runtimeTable = `
+        CREATE TABLE runtime(
+            Id INTEGER NOT NULL PRIMARY KEY,
+            SchemaVersion INTEGER NOT NULL,
+            StaticDir TEXT NOT NULL,
+            TmpDir TEXT NOT NULL,
+            RunRoot TEXT NOT NULL,
+            GraphRoot TEXT NOT NULL,
+            GraphDriverName TEXT NOT NULL,
+            CHECK (Id=0)
+        );
+        `
+	const fillRuntimeTable = `INSERT INTO runtime VALUES (
+            ?, ?, ?, ?, ?, ?, ?
+        );`
+
+	const selectRuntimeTable = `SELECT SchemaVersion,
+                                           StaticDir,
+                                           TmpDir,
+                                           RunRoot,
+                                           GraphRoot,
+                                           GraphDriverName
+                                    FROM runtime WHERE id=0;`
+
+	const checkRuntimeExists = "SELECT name FROM sqlite_master WHERE type='table' AND name='runtime';"
+
+	tx, err := db.Begin()
+	if err != nil {
+		return errors.Wrapf(err, "error beginning database transaction")
+	}
+	defer func() {
+		if err != nil {
+			if err2 := tx.Rollback(); err2 != nil {
+				logrus.Errorf("Error rolling back transaction to check runtime table: %v", err2)
+			}
+		}
+
+	}()
+
+	row := tx.QueryRow(checkRuntimeExists)
```

Comment:
Too much code today.  When you move from  version 1 to version 2, will this code update itself appropriately?  We need to ensure that there is only one row in the table with the latest schema version I think.

Reply:
There's a constraint in the database to ensure that there is only ever one row (the ID/primary key is forced to 0). We will never update this row, mainly because we don't have a good way to migate between database revisions, so for now if the schema changes this will just give you a nice, loud notice that you need to delete it and recreate it.

Reply:
Thanks for the 411.  Just to be 100% sure, this notice will happen during one of our CI runs after a particular submit rather than on a customer's system right?

Reply:
It would happen on customer systems. We need to investigate proper schema migration at some point, definitely before we merge into CRI-O. This just clarifies errors so we know what's blowing up.